pkg/liqonet/ipam_test.go

```diff
@@ -87,42 +109,95 @@ var _ = Describe("IpamNew", func() {
 						_, err = ipam.GetSubnetPerCluster("11.0.0.0/16", "cluster4")
 						gomega.Expect(err).ToNot(gomega.BeNil())
-						err = ipam.FreeSubnetPerCluster("cluster1")
-						gomega.Expect(err).To(gomega.BeNil())
-						err = ipam.FreeSubnetPerCluster("cluster2")
-						gomega.Expect(err).To(gomega.BeNil())
-						err = ipam.FreeSubnetPerCluster("cluster3")
-						gomega.Expect(err).To(gomega.BeNil())
 					})
 				})
 			})
 		})
-		Context("When the remote cluster asks for a subnet belonging to a network in the pool", func() {
-			Context("and all pools are full", func() {
-				_, err := ipam.GetSubnetPerCluster("10.0.0.0/9", "cluster1")
-				gomega.Expect(err).To(gomega.BeNil())
-				gomega.Expect(ipam.SubnetPerCluster["cluster1"]).To(gomega.Equal("10.0.0.0/9"))
-				_, err = ipam.GetSubnetPerCluster("10.128.0.0/9", "cluster2")
-				gomega.Expect(err).To(gomega.BeNil())
-				gomega.Expect(ipam.SubnetPerCluster["cluster2"]).To(gomega.Equal("10.128.0.0/9"))
-				_, err = ipam.GetSubnetPerCluster("10.1.0.0/16", "cluster3")
-				gomega.Expect(err).ToNot(gomega.BeNil())
-				err = ipam.FreeSubnetPerCluster("cluster1")
+		Context("When the remote clusters asks for a subnet which is equal to a pool", func() {
+			BeforeEach(func() {
+				var ipam *liqonet.IPAM
+				pool := []string{
+					"10.0.0.0/8",
+					"172.16.0.0/12",
+				}
+				reserved := []string{
+					"192.168.1.0/24",
+				}
+				subnetPerCluster := make(map[string]string)
+				ipam = liqonet.NewIPAM()
+				err := ipam.Init(reserved, pool, subnetPerCluster)
 				gomega.Expect(err).To(gomega.BeNil())
-				err = ipam.FreeSubnetPerCluster("cluster2")
+			})
+			It("should map it to another network", func() {
+				network, err := ipam.GetSubnetPerCluster("172.16.0.0/12", "cluster1")
 				gomega.Expect(err).To(gomega.BeNil())
+				gomega.Expect(network).ToNot(gomega.Equal("172.16.0.0/12"))
+			})
+		})
+		Context("When the remote cluster asks for a subnet belonging to a network in the pool", func() {
+			Context("and all pools are full", func() {
+				BeforeEach(func() {
+					pool := []string{
+						"10.0.0.0/8",
+					}
+					reserved := []string{
+						"192.168.1.0/24",
+					}
+					subnetPerCluster := make(map[string]string)
+					ipam = liqonet.NewIPAM()
+					err := ipam.Init(reserved, pool, subnetPerCluster)
+					gomega.Expect(err).To(gomega.BeNil())
+				})
+				It("should not allocate the network", func() {
+					_, err := ipam.GetSubnetPerCluster("10.0.0.0/9", "cluster1")
+					gomega.Expect(err).To(gomega.BeNil())
+					gomega.Expect(ipam.SubnetPerCluster["cluster1"]).To(gomega.Equal("10.0.0.0/9"))
+					_, err = ipam.GetSubnetPerCluster("10.128.0.0/9", "cluster2")
+					gomega.Expect(err).To(gomega.BeNil())
+					gomega.Expect(ipam.SubnetPerCluster["cluster2"]).To(gomega.Equal("10.128.0.0/9"))
+					_, err = ipam.GetSubnetPerCluster("10.1.0.0/16", "cluster3")
+					gomega.Expect(err).ToNot(gomega.BeNil())
+					/*err = ipam.FreeSubnetPerCluster("cluster1")
```

Comment:
Commented code is dead code and should be dropped or decommented

Reply:
Sorry, I missed it!